src/Compilers/Core/Portable/Compilation/ConversionInfo.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Semantics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Summarizes whether a conversion is allowed, and if so, which kind of conversion (and in 
+    /// some cases, the associated symbol).
+    /// </summary>
+    public struct ConversionInfo : IEquatable<ConversionInfo>
```

Comment:
I could not use a single enum because of the need to store the IMethodSymbol.  As such, a struct felt the most natural.  

Open questions:

While this covers a wide range of conversion types, there are a few conversion types i could not fit in *yet* due to not being quite sure if they would make sense across VB and C#.

Here were the cases i have not covered yet:

1. C# has Boxing/Unboxing conversions.  These are interesting for a few reasons.  First, we have Widening/Narrowing conversions below.  These tend to pair up with the rest.  I.e. Widening-Reference, Narrowing-Reference.  Widening-Numeric, Narrowing-Numeric.  It feels like there shouldn't be Boxing/Uboxing.  But instead there should be Widening-Boxing, and Narrowing-Boxing.   Second, the C# compiler uses Boxing both for boxing value-types, as well as for boxing type-parameters.   Vb calls out ValueType and TypeParameter conversions separately.   I feel like the right design would be to have IsValueType and IsTypeParameter, and perhaps provide IsBoxing()/IsUnboxing() extension methods that are a combination of "(IsValueType || IsTypeParameter) && IsWidening).  Thoughts?
2. C# has IsAnonymousFunction, whereas VB has IsLambda and IsAnonymousDelegate.  It's unclear how these should be best be exposed.  
3. C# has IsEnumeration.  VB has no such concept.
4. C# has IsMethodGroup.  VB has no such concept.
5. C# has IsInterpolatedString, Vb has IsString.  Should these be exposed as one thing?
6. C# has IsDynamic.  VB has no such concept.
7. C# has tuple conversion info.  Vb does not expose that currently.
8. VB has IsArray.  C# does not.